eng/pipelines/templates/jobs/live.tests.yml

```diff
@@ -164,32 +173,60 @@ jobs:
       SubscriptionConfiguration: $(SubscriptionConfiguration)
       EnvVars:
         Pool: $(Pool)
+        ${{ insert }}: ${{ parameters.EnvVars }}
+      UseFederatedAuth: ${{ parameters.UseFederatedAuth }}
+      ServiceConnection: ${{ parameters.CloudConfig.ServiceConnection }}
 
   - template: /eng/common/testproxy/test-proxy-tool.yml
     parameters:
       runProxy: false
 
   - ${{ parameters.PreTestSteps }}
 
-    # For non multi-config generator use the same build configuration to run tests
-    # We don't need to set it to invoke ctest
-    # Visual Studio generator used in CI is a multi-config generator.
-    # As such, it requires the configuration argument for building and invoking ctest
-  - bash: |
-      export AZURE_CLIENT_ID=$(${{parameters.ServiceDirectory}}_CLIENT_ID)
-      export AZURE_TENANT_ID=$(${{parameters.ServiceDirectory}}_TENANT_ID)
-      export AZURE_CLIENT_SECRET=$(${{parameters.ServiceDirectory}}_CLIENT_SECRET)
+  - ${{ if parameters.UseFederatedAuth }}:
+    - task: AzurePowerShell@5
+      displayName: ctest
+      # Runs only if test-resources are happily deployed.
+      # unit-tests runs for those configs where samples are not ran.
+      # This enables to run tests and samples at the same time as different matrix configuration.
```

Comment:
Are these comments up-to-date? What is "unit-tests" being referred to? And what's the relationship between this yml and when samples run or not?

Reply:
I think these are relatively up to date but also not totally necessary anymore.